src/Sarif.Converters/FxCopConverter.cs

```diff
@@ -85,8 +85,19 @@ internal Result CreateResult(FxCopLogReader.Context context)
 
             if (!String.IsNullOrWhiteSpace(uniqueId))
             {
-                result.ToolFingerprintContributions = new Dictionary<string, string>();
-                result.ToolFingerprintContributions.Add("UniqueId", uniqueId);
+                if (result.ToolFingerprintContributions == null)
+                {
+                    result.ToolFingerprintContributions = new Dictionary<string, string>();
+                }
+
+                if (result.ToolFingerprintContributions.ContainsKey("UniqueId"))
+                {
+                    result.ToolFingerprintContributions["UniqueId"] = uniqueId;
+                }
+                else
+                {
+                    result.ToolFingerprintContributions.Add("UniqueId", uniqueId);
+                }
```

Comment:
This is fine. After this PR, consider adding a helper method
```
static void SafeAddDictionaryElement(Dictionary<TKey, TValue> dict, TKey key, TValue value)
```
... to some `Utilities` class to encapsulate this repeated logic. Perhaps in the `SarifUtilities` class.